File: bag.m
% Defines the properties a bag that can contain items 
classdef bag
    properties 
        position int32
        % 1x2 integer array = coordinates 
        rotation int32
        % 1=90degrees 2=180degrees etc
        blocks int32
        % 2xN array of relative coordinates relative to origin
        bag_type
        % Decides which other items can be modified by this bag
    end
end

File: cpp/src/main/bag.cpp
#include <vector>
#include <array>

class Bag {
    public:
        // 1x2 integer array = coordinates 
        std::array<int, 2> position;
        // 1=90degrees 2=180degrees etc
        int rotation;
        // vector of coordinates relative to origin
        std::vector<std::array<int, 2>> blocks;
        // Decides which other items can be modified by this bag
        int bag_type;

        Bag(std::vector<std::array<int, 2>> blocks)
          : blocks(blocks)
        {}
};